grails-app/services/org/pih/warehouse/core/LocationService.groovy

```diff
@@ -169,16 +173,21 @@ class LocationService {
 			List<Location> internalLocations = Location.createCriteria().list() {
 				eq("active", Boolean.TRUE)
 				eq("parentLocation", parentLocation)
-				locationType {
-					'in'("locationTypeCode", locationTypeCodes)
+				or {
+					locationType {
+						'in'("locationTypeCode", locationTypeCodes)
+					}
+					if (locationName) {
+						eq("name", locationName)
+					}
 				}
 			}
 
 			// Filter by activity code
 			if (activityCodes) {
 				activityCodes.each { activityCode ->
 					internalLocations = internalLocations.findAll { internalLocation ->
-						internalLocation.supports(activityCode)
+						internalLocation.supports(activityCode) || (locationName && internalLocation.name == locationName)
```

Comment:
I don't love this, but understand why you needed to do it. This wouldn't be necessary if we could filter by activity code in the database query, so maybe I'll look into refactoring the query to handle activity code as well. The issue is that the location inherits activity codes from the location type so it's a little more complicated than 
```'in'("supportedActivities", activityCodes)```